backend/app/services/knowledge/code_wiki/generation.py

```
@@ -208,7 +213,21 @@ def start_generation(
     # seeded versions, and whichever finishes last silently overwrites the other.
     # The knowledge base row always exists, so locking it serialises starts per wiki
     # without depending on gap-lock behaviour that the isolation level can turn off.
-    db.query(Kind).filter(Kind.id == knowledge_base.id).with_for_update().first()
+    locked_knowledge_base = (
+        db.query(Kind)
+        .filter(
+            Kind.id == knowledge_base.id,
+            Kind.kind == "KnowledgeBase",
+            Kind.is_active == True,
```

Ruff (0.16.1)

[error] Avoid equality comparisons to `True`; use `Kind.is_active:` for truth checks

Replace with `Kind.is_active`

(E712)